org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/jvmmodel/DomainmodelJvmModelHelper.xtend

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.example.domainmodel.jvmmodel
+
+import com.google.common.collect.HashMultimap
+import com.google.common.collect.Multimap
+import com.google.inject.Inject
+import java.util.Collection
+import java.util.function.Consumer
+import org.eclipse.xtext.common.types.JvmDeclaredType
+import org.eclipse.xtext.common.types.JvmOperation
+import org.eclipse.xtext.xbase.typesystem.^override.OverrideHelper
+
+/**
+ * @author Lorenzo Bettini - Initial contribution and API, https://github.com/eclipse/xtext-eclipse/issues/1205
+ */
+class DomainmodelJvmModelHelper {
+
+	@Inject extension OverrideHelper
+
+	/**
+	 * Detects duplicated {@link JvmOperation}s in the passed {@link JvmDeclaredType}
+	 * taking into consideration overloading and type erasure; each collection of
+	 * duplicates is passed to the consumer.
+	 */
+	def void handleDuplicateJvmOperations(JvmDeclaredType jvmDeclaredType,
+		Consumer<Collection<JvmOperation>> consumer) {
+		// takes into consideration overloading and type erasure
+		val methods = jvmDeclaredType.getResolvedFeatures.declaredOperations
+		val Multimap<String, JvmOperation> signature2Declarations = HashMultimap.create
```

Comment:
If I understand correctly how Multimaps.index works (values and lambda for computing keys), that's not applicable in this context: starting from `IResolvedOperation`s I need to map strings to `JvmOperation`s and the string is computed from `IResolvedOperation` instead of from `JvmOperation`. Or am I missing something?

Reply:
You are right.